### PR TITLE
Eslint update

### PR DESCRIPTION
**Pull request by lrowe**
Update to eslint 2. The first four commits fix consistent-return linting errors.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -43,8 +43,8 @@
         "consistent-return": [ 2 ],
         "curly": [ 2 ],
         "default-case": [ 2 ],
-        "dot-notation": [ [ 2 ], { "allowKeywords": true } ],
-        "eqeqeq": [ 2 ],
+        "dot-notation": [ 2, { "allowKeywords": true } ],
+        "eqeqeq": [ 2, "smart" ],
```
Comment (yunong) [inline .eslintrc]: Why not just use === across the board?

Comment (lrowe) [inline .eslintrc]: Just the relatively large number of errors it throws up (26.)

Comment (yunong) [inline .eslintrc]: Is it worth while fixing? Seems like an easy fix for literals.

Comment (lrowe) [inline .eslintrc]: I'd like to avoid the work of sorting out which of the various `!= null` comparisons need to also take account of undefined.

**Pull request merged**